server/src/arcs-master-app.ts

```diff
@@ -32,41 +40,128 @@ class ArcsMasterApp extends AppBase {
    * @param str a base64 string, usually a fingerprint.
    */
   gcpSafeIdentifier(str) {
-    return str.replace(/[ +-=]/g, '').toLowerCase().substring(0, 32);
+    return str.replace(/[ +\-=]/g, '').toLowerCase().substring(0, 32);
   }
 
-  async findDeployment(req, res, next) {
+  async lock(req, res, next) {
+    console.log("fingerprint is " + req.params.fingerprint);
     const fingerprint = this.gcpSafeIdentifier(req.params.fingerprint);
+    console.log("gcp safe fingerprint " + fingerprint);
+    try {
+      const cloud = CloudManager.forGCP();
+      const disk = await cloud.disks().find(fingerprint);
+      if (disk) {
+        const attached = await disk.isAttached();
+
+        if (attached) {
+          const container: Container | null = await cloud.containers().find(fingerprint);
+          if (container != null) {
```

Comment:
Perhaps we should log something here (and for unlock case below) if the container is null? Since the disk will not be dismounted, but seems not to in fact be attached to the noted container (or, clarify if I mis-speak).

I see you have a TODO at line 89 below for the unlock case, so I'm wondering what we should do in this case, and probably we should log something in unlock as well.